src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt

```diff
@@ -49,56 +49,63 @@ class CourtEventProcessor(
       sqsMessage.message,
     )
 
-    val uniqueDefendants = commonPlatformHearingEvent.hearing.prosecutionCases
-      .flatMap { it.defendants }
-      .filterNot { it.isYouth }
-      .distinctBy {
-        it.personDefendant?.personDetails?.firstName +
-          it.personDefendant?.personDetails?.lastName +
-          it.personDefendant?.personDetails?.dateOfBirth +
-          it.pncId +
-          it.cro
+    run {
+      val uniqueDefendants = commonPlatformHearingEvent.hearing.prosecutionCases
+        .flatMap { it.defendants }
+        .filterNot { it.isYouth }
+        .distinctBy {
+          it.personDefendant?.personDetails?.firstName +
+            it.personDefendant?.personDetails?.lastName +
+            it.personDefendant?.personDetails?.dateOfBirth +
+            it.pncId +
+            it.cro
+        }
+      val defendantIDs = uniqueDefendants.joinToString(" ") { it.id.toString() }
+      log.debug("Processing Common Platform Event with ${uniqueDefendants.size} distinct defendants with defendantId $defendantIDs")
+
+      uniqueDefendants.forEach { defendant ->
+        val person = Person.from(defendant)
+        telemetryService.trackEvent(
+          MESSAGE_RECEIVED,
+          mapOf(
+            EventKeys.DEFENDANT_ID to person.defendantId,
+            EventKeys.EVENT_TYPE to COMMON_PLATFORM_HEARING.name,
+            EventKeys.MESSAGE_ID to sqsMessage.messageId,
+            EventKeys.SOURCE_SYSTEM to SourceSystemType.COMMON_PLATFORM.name,
+          ),
+        )
+        createUpdateService.processMessage(person) {
+          person.defendantId?.let {
+            personRepository.findByDefendantId(it)
+          }
+        }
+        person.defendantId = null
```

Comment:
What is the reason for this? this wont effect any thing as it is set after processing